guides/common/modules/proc_disabling-puppet.adoc

```diff
@@ -17,8 +17,8 @@ To discontinue using Puppet in your {Project}, follow this procedure.
 +
 [options="nowrap" subs="+quotes,attributes"]
 ----
-# {foreman-maintain} plugin purge-puppet [--remove-all-data]
+# {foreman-maintain} plugin purge-puppet --remove-all-data
 ----
 +
-The command without arguments will remove all Puppet-related data in {Project}.
-To remove Puppet server data files, including Puppet environments, add the `--remove-all-data` argument to the command.
+The command without arguments will remove all Puppet-related data in {Project} database.
```

Comment:
If all Puppet-related data does not include Puppet environments, shouldn't this be reworded?

Reply:
That's why I added the "database" at the end. The argument will remove "files". (I assume.)

Reply:
https://github.com/theforeman/foreman_maintain/blob/master/definitions/procedures/puppet/remove_puppet_data.rb only gets executed if `--remove-all-data` is passed. So effectively `foreman-rake purge:puppet && rm -rf /etc/puppetlabs /opt/puppetlabs/server/data`. It always attempts to clean the DB by rolling back DB migrations from the plugin, but removing the files on disk only happens with `--remove-all-data`. The `foreman-rake purge:puppet` task can also remove the DB tables and columns without the plugin installed, so is there to really make sure it works.

Reply:
@ekohl So that means it's correct in the docs now, right?

Reply:
I'd say "without arguments" can be interpreted as simply running `{foreman-maintain}` since `plugin purge-puppet` are also arguments.

What's also incorrect is that the same command can be used (both with and without `--remove-all-data`) on both Server and Proxies. The current text implies `--remove-all-data` can only be used on Server.

It is required that you first run `plugin purge-puppet` (or `plugin purge-puppet --remove-all-data`) on all Proxies where Puppet was enabled. Once that's done, run it on Server. If there is still a Proxy with Puppet enabled, the purge command will fail. It's supposed to refuse to run, so no harm should be done, but we shouldn't set up users to run into that scenario.

Does that help on what should be achieved?

Reply:
@ekohl Well, we're telling them to disable and remove Puppet on all SmartProxies in the previous step. Do you mean there also should be the `--remove-all-data` argument in the command in the previous step?